### PR TITLE
[EMCAL-767]: Add histograms for # of good, bad, dead cells in EMCAL/DCAL.

### PR DESCRIPTION
**Pull request by hjbossi**

### CODE REVIEW

**Comment by hjbossi** [comment]
@mfasDa @jokonig to check

File: Modules/EMCAL/src/CalibMonitoringTask.cxx [hunk 1]
```diff
@@ -68,6 +68,22 @@ void CalibMonitoringTask::initialize(Trigger, framework::ServiceRegistryRef)
       mBadChannelMapHisto->GetYaxis()->SetTitle("row (#phi)");
       mBadChannelMapHisto->SetStats(false);
       getObjectsManager()->startPublishing(mBadChannelMapHisto);
+
+      // histogram for number of bad, dead, good channels in emcal only
+      mBadChannelMapEMCALHisto = new TH2D("badChannelMapEMCAL", "Number of Good/Dead/Bad Channels in EMCAL Only", 3, 0, 2, 12288, 0, 12288);
```
Comment (mfasDa) [inline Modules/EMCAL/src/CalibMonitoringTask.cxx:73]: Maybe use a more specific histogram name, like maskStatsEMCAL/DCAL.

File: Modules/EMCAL/src/CalibMonitoringTask.cxx [hunk 2]
```diff
@@ -68,6 +68,22 @@ void CalibMonitoringTask::initialize(Trigger, framework::ServiceRegistryRef)
...
+      mBadChannelMapEMCALHisto->GetXaxis()->SetTitle("channel status");
+      mBadChannelMapEMCALHisto->GetYaxis()->SetTitle("Number of channels");
+      mBadChannelMapEMCALHisto->SetStats(false);
+      getObjectsManager()->startPublishing(mBadChannelMapEMCALHisto);
+
+      // histogram for number of bad, dead, good channels in emcal only
+      mBadChannelMapDCALHisto = new TH2D("badChannelMapDCAL", "Number of Good/Dead/Bad Channels in DCAL Only", 3, 0, 2, 5376, 0, 5376);
```
Comment (mfasDa) [inline Modules/EMCAL/src/CalibMonitoringTask.cxx:80]: Same as above.

File: Modules/EMCAL/src/CalibMonitoringTask.cxx [hunk 3]
```diff
@@ -88,6 +104,14 @@ void CalibMonitoringTask::update(Trigger t, framework::ServiceRegistryRef)
...
         for (Int_t j = 0; j < hist_temp2->GetNbinsY(); j++) {
           mBadChannelMapHisto->SetBinContent(i + 1, j + 1, hist_temp2->GetBinContent(i + 1, j + 1));
+          // check to see if the cell is in the emcal or dcal and fill appropriate hist
+          int cellID = geo->GetAbsCellIdFromCellIndexes(i, j);
+          if (cellID < 12288) {
+            mBadChannelMapEMCALHisto->Fill(hist_temp2->GetBinContent(i + 1, j + 1));
+          }
+          else{
+            mBadChannelMapDCALHisto->Fill(hist_temp2->GetBinContent(i + 1, j + 1));
+          }
```
Comment (mfasDa) [inline Modules/EMCAL/src/CalibMonitoringTask.cxx:114]: There is also a pointer to the bad channel map itself, you would not need the geometry functions, you know the tower ID. Iterating over the bad channel map is super fast, calculating the the cell indices is potentially expensive, a separate for loop makes sense for performance reasons.

**Review by mfasDa — CHANGES_REQUESTED** [review]
Few things to be changed. In particular I think it is computing wise more advantageous to use a separate loop over the channels for filling the stat histograms, because they are potentially expensive.

**Review by knopers8 — APPROVED** [review]

**Comment by mfasDa** [comment]
CI errors now unrelated

**Pull request merged**